packages/ag-charts-community/src/scene/shape/text.ts

```diff
@@ -94,7 +94,16 @@ export class Text<D = any> extends Shape<D> {
         const { fill, stroke, strokeWidth } = this;
         const { pixelRatio } = this.layerManager.canvas;
 
-        ctx.font = TextUtils.toFontString(this);
+        if (!fill && !(stroke != null && strokeWidth > 0)) {
+            // Short circuit early if nothing will be rendered.
+            return super.render(renderCtx);
+        }
+
+        const font = TextUtils.toFontString(this);
+        // Try to avoid this assignment, which typically always incurs a font switch cost.
+        if (ctx.font !== font) {
```

Comment:
If this is showing as a hot path, it might be worth storing the resolved font on the instance too, and reading that to avoid the overhead of crossing the language barrier since the `.font` getter is C++

Reply:
This is probably ok, as with a prior iteration of this change the hotspot was significantly reduced with just this guard added. We'd need to track the current font in the renderContext for this to be accurate, which seems more complexity than I'd like to add right now.